helpers/class.TestSessionStorage.php

```diff
@@ -131,17 +131,14 @@ public function setUserUri($userUri)
      */
     public function retrieve(AssessmentTest $test, $sessionId, $forReadingOnly = false)
     {
-        if (empty(static::$session) || static::$session->getSessionId() !== $sessionId) {
-            $this->setLastError(-1);
-            static::$session = parent::retrieve($test, $sessionId);
-            static::$session->setReadOnly($forReadingOnly);
-            if (!$forReadingOnly) {
-                $this->lockSession(static::$session);
-            }
+        if (static::$session && static::$session->getSessionId() === $sessionId) {
+            return static::$session;
```

Comment:
if first call of this method will be with `$forReadingOnly = true` the second call with `$forReadingOnly = false` will not lock the session.

Reply:
This is the case which happens when proctor tries to reactivate terminated session. At first session is loaded in read only mode and later when we set state we change more to `readOnly = false`. I tested it, worked fine.

Reply:
Fixed